packages/kbn-legacy-logging/src/log_reporter.ts

```diff
@@ -31,21 +33,28 @@ export function getLogReporter({ events, config }: { events: any; config: LogFor
   const format = config.json ? new KbnLoggerJsonFormat(config) : new KbnLoggerStringFormat(config);
   const logInterceptor = new LogInterceptor();
 
-  let dest: WriteStream | NodeJS.WritableStream;
   if (config.dest === 'stdout') {
-    dest = process.stdout;
+    pipeline(logInterceptor, squeeze, format, onFinished);
+    // The `pipeline` function is used to properly close all streams in the
+    // pipeline in case one of them ends or fails. Since stdout obviously
+    // shouldn't be closed in case of a failure in one of the other streams,
+    // we're not including that in the call to `pipeline`, but rely on the old
+    // `pipe` function instead.
+    format.pipe(process.stdout);
```

Comment:
I know that this code is not tested, but would you mind adding a single test for added functionality?

Reply:
Done 👍